lib/taskana-core/src/main/java/pro/taskana/spi/task/api/BeforeRequestReviewProvider.java

```diff
@@ -0,0 +1,43 @@
+package pro.taskana.spi.task.api;
+
+import pro.taskana.common.api.TaskanaEngine;
+import pro.taskana.task.api.models.Task;
+
+/**
+ * The BeforeRequestReviewProvider allows to implement customized behaviour before a review has been
+ * requested on a given {@linkplain Task}.
+ */
+public interface BeforeRequestReviewProvider {
+
+  /**
+   * Provide the active {@linkplain TaskanaEngine} which is initialized for this TASKANA
+   * installation.
+   *
+   * <p>This method is called during TASKANA startup and allows the service provider to store the
+   * active {@linkplain TaskanaEngine} for later usage.
+   *
+   * @param taskanaEngine the active {@linkplain TaskanaEngine} which is initialized for this
+   *     installation.
+   */
+  void initialize(TaskanaEngine taskanaEngine);
+
+  /**
+   * Perform any action before a review has been requested on a {@linkplain Task} through
+   * {@linkplain pro.taskana.task.api.TaskService#requestReview(String)}.
+   *
+   * <p>This SPI is executed within the same transaction staple as {@linkplain
+   * pro.taskana.task.api.TaskService#requestReview(String)}.
+   *
+   * <p>This SPI is executed with the same {@linkplain
+   * pro.taskana.common.api.security.UserPrincipal} and {@linkplain
+   * pro.taskana.common.api.security.GroupPrincipal} as in {@linkplain
+   * pro.taskana.task.api.TaskService#requestReview(String)}.
+   *
+   * @param task the {@linkplain Task} before {@linkplain
+   *     pro.taskana.task.api.TaskService#requestReview(String)} has started
+   * @return the modified {@linkplain Task}. <b>IMPORTANT:</b> persistent changes to the {@linkplain
+   *     Task} have to be managed by the service provider.
```

Comment:
Shouldn't the changes be persistent if the transaction in requestReview is successful? Can we choose a better wording?

Reply:
Please remove the point

Reply:
point removed.

Reply:
> Shouldn't the changes be persistent if the transaction in requestReview is successful? Can we choose a better wording?
You got any suggestions? What we want to say that modification of the Task should be called by and TaskService method.
This applies to the other SPIs aswell